### DISCUSSION TITLE
Why won't env work in task.json?

### DISCUSSION THREAD

**Discussion by garmede** [microsoft/vscode-cpptools#7635] [Q&A]
I tried excute task like below:
```json
{
    "version": "2.0.0",
    "options": {
        "env": {
            "VCPKG_ROOT": "F:\\GitHub\\vcpkg\\installed"
        }
    },
    "tasks": [
        {
            "label": "command test",
            "type": "cppbuild",
            "command": "echo",
            "args": [
                "${env:VCPKG_ROOT}"
            ],
            "group": {
                "kind": "build",
                "isDefault": true
            },
            "problemMatcher": []
        }
    ]
}
```
Then print Error
```shell
echo 
ECHO�� �����Ǿ� �ֽ��ϴ�.
```

Why don't use `env` option  in `"type": "cppbuild"`?

But works good in `"type": "shell"`.

**Comment by sean-mcmanus** [comment]
That seems like it could be a VS Code bug with the CustomExecution class, which handles all the variable resolving for the task. Can you file a VS Code bug? You could refer them to our usage at https://github.com/microsoft/vscode-cpptools/blob/main/Extension/src/LanguageServer/cppBuildTaskProvider.ts#L199 .
